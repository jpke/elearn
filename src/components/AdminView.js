import React from 'react';
import {Link} from 'react-router';

const AdminView = (props) => (
  <div className="admin-container">
    <Link to="/" className="toggleView">
      <button id="toggleView" className="toggleView">Back to Courses
      </button>
    </Link>
    <div className="edit-course-header">
      <div>
      </div>
      <div className="title">
        <h3>Title</h3>
        <input type="text" id="course-title" name="course-title" className="edit-course-title" value={props.course.name} onChange={(e) => {props.editCourse(e)}}/>
      </div>
      <div className="save">
        <button id="updateCourse" className="updateCourse" onClick={() => {props.updateCourse(props.token, props.course)}}>Save</button>
      </div>
    </div>
    <div>
      <h3>Email addresses of students who can register for the course</h3>
    </div>
    <form name="newUserForm" onSubmit={(e) => {props.addUser(e)}}>
      <input type='email' id="new-user" name="newUser" defaultValue={props.newUser} className="edit-new-user" placeholder='newUser@email.com' />
      <button className="add-student" type='submit'>Add Student</button>
    </form>
    <div className="enrollable-users">
      <ul className="user-list">
        {props.listUsers(props.course.enrollable, props.deleteUser)}
      </ul>
    </div>
  </div>
);

export default AdminView;
