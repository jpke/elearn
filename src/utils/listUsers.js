import React from 'react'

//creates list of users enrollable or enrolled in course
//includes option to delete selected user
const listUsers = (users, deleteUser) => {
  if(users) {
    return users.map((user, index) => {
      return (
        <li key={index} className="listed-item">
          <div>{user.email}</div>
          {user.admin && <div>Admin</div>}
          <button onClick={() => {deleteUser(user.email)}}>Delete</button>
        </li>
        )
    })
  }
}

export default listUsers;
