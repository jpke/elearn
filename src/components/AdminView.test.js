/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

import AdminView from './AdminView';
import listUsers from '../utils/listUsers';

let editCourse, course;
beforeEach(() => {
  editCourse = jest.fn();
  course = {
    _id: "0000",
    name: "example course"
  };
})

it('renders correctly initally', () => {
  const tree = renderer.create(
    <AdminView
      editCourse={jest.fn()}
      addUser={jest.fn()}
      deleteUser={jest.fn()}
      deleteUserFromCourse={jest.fn()}
      message={""}
      updateCourse={jest.fn()}
      listUsers={listUsers}
      newUser={""}
      course={""}
      enrollable={[]}
      enrolled={[]}
      _id={""}
      token={""}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly with enrolled users', () => {
  const tree = renderer.create(
    <AdminView
      editCourse={jest.fn()}
      addUser={jest.fn()}
      deleteUser={jest.fn()}
      deleteUserFromCourse={jest.fn()}
      message={""}
      updateCourse={jest.fn()}
      listUsers={listUsers}
      newUser={"testNew"}
      course={course}
      enrollable={[{email: "new@email.com"}]}
      enrolled={[{email: "enrolled@email.com"}]}
      _id={"00000"}
      token={"someToken"}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('triggers input onChange callback', () => {
  const wrapper = shallow(
    <AdminView
      editCourse={editCourse}
      addUser={jest.fn()}
      deleteUser={jest.fn()}
      deleteUserFromCourse={jest.fn()}
      message={""}
      updateCourse={jest.fn()}
      listUsers={listUsers}
      newUser={"testNew"}
      course={course}
      enrollable={[{email: "new@email.com"}]}
      enrolled={[{email: "enrolled@email.com"}]}
      _id={"00000"}
      token={"someToken"}
      />
  );
  expect(jest.fn()).not.toHaveBeenCalled();
  wrapper.find('.edit-course-title').simulate('change', "example course edit")
  expect(editCourse).toHaveBeenCalledWith("example course edit");
})
