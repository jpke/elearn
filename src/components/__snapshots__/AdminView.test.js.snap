exports[`test renders correctly initally 1`] = `
<div
  className="admin-container">
  <a
    className="toggleView"
    onClick={[Function]}
    style={Object {}}>
    <button
      className="toggleView"
      id="toggleView">
      Back to Courses
    </button>
  </a>
  <div>
    <h3
      className="message">
      
    </h3>
  </div>
  <div
    className="edit-course-header">
    <div />
    <div
      className="title">
      <h3>
        Title
      </h3>
      <input
        className="edit-course-title"
        id="course-title"
        name="course-title"
        onChange={[Function]}
        type="text"
        value={undefined} />
    </div>
    <div
      className="save">
      <button
        className="updateCourse"
        id="updateCourse"
        onClick={[Function]}>
        Save
      </button>
    </div>
  </div>
  <div>
    <h3>
      Email addresses of students who can register for the course
    </h3>
  </div>
  <form
    name="newUserForm"
    onSubmit={[Function]}>
    <div>
      <input
        className="edit-new-user"
        defaultValue=""
        id="new-user"
        name="newUser"
        placeholder="newUser@email.com"
        type="email" />
      <div
        className="admin-checkbox">
        <label>
          Admin:   
        </label>
        <input
          className="edit-new-user-admin"
          id="new-user-admin"
          name="admin"
          type="checkbox" />
      </div>
    </div>
    <button
      className="add-student"
      type="submit">
      Add Student
    </button>
  </form>
  <div
    className="enrollable-users">
    <h4>
      Users who can enroll
    </h4>
    <ul
      className="user-list" />
  </div>
  <div
    className="enrolled-users">
    <h4>
      Enrolled users
    </h4>
    <ul
      className="user-list-enrolled" />
  </div>
</div>
`;

exports[`test renders correctly with enrolled users 1`] = `
<div
  className="admin-container">
  <a
    className="toggleView"
    onClick={[Function]}
    style={Object {}}>
    <button
      className="toggleView"
      id="toggleView">
      Back to Courses
    </button>
  </a>
  <div>
    <h3
      className="message">
      
    </h3>
  </div>
  <div
    className="edit-course-header">
    <div />
    <div
      className="title">
      <h3>
        Title
      </h3>
      <input
        className="edit-course-title"
        id="course-title"
        name="course-title"
        onChange={[Function]}
        type="text"
        value={undefined} />
    </div>
    <div
      className="save">
      <button
        className="updateCourse"
        id="updateCourse"
        onClick={[Function]}>
        Save
      </button>
    </div>
  </div>
  <div>
    <h3>
      Email addresses of students who can register for the course
    </h3>
  </div>
  <form
    name="newUserForm"
    onSubmit={[Function]}>
    <div>
      <input
        className="edit-new-user"
        defaultValue="testNew"
        id="new-user"
        name="newUser"
        placeholder="newUser@email.com"
        type="email" />
      <div
        className="admin-checkbox">
        <label>
          Admin:   
        </label>
        <input
          className="edit-new-user-admin"
          id="new-user-admin"
          name="admin"
          type="checkbox" />
      </div>
    </div>
    <button
      className="add-student"
      type="submit">
      Add Student
    </button>
  </form>
  <div
    className="enrollable-users">
    <h4>
      Users who can enroll
    </h4>
    <ul
      className="user-list">
      <li
        className="listed-item">
        <div>
          new@email.com
        </div>
        <button
          onClick={[Function]}>
          Delete
        </button>
      </li>
    </ul>
  </div>
  <div
    className="enrolled-users">
    <h4>
      Enrolled users
    </h4>
    <ul
      className="user-list-enrolled">
      <li
        className="listed-item">
        <div>
          enrolled@email.com
        </div>
        <button
          onClick={[Function]}>
          Delete
        </button>
      </li>
    </ul>
  </div>
</div>
`;
